core/src/main/java/org/infinispan/scattered/impl/ScatteredStateConsumerImpl.java

```diff
@@ -105,6 +105,10 @@ protected void beforeTopologyInstalled(int topologyId, boolean startRebalance, C
       // and ScatteringInterceptor would check according to the new CH.
       for (int segment = 0; segment < newWriteCh.getNumSegments(); ++segment) {
          if (!newWriteCh.isSegmentLocalToNode(rpcManager.getAddress(), segment)) {
+            // We cancel any existing transfers as failed key transfer could move segment
+            // to OWNED state if it fails, concurrently with installing the topology.
```

Comment:
If the key transfer fails, we assume that the node has crashed. This node then has to take the ownership, there's not much else to do when we've lost data.
If the key transfer is cancelled, it is because there is a newer topology installed and we don't seize the ownership, it's up to the new topology to select who will be the next owner.

Key transfer is not installing a topology. When a rebalance starts, all new segments are marked as `BLOCKED` and we wait until the previous owner confirms that it has revoked them. Then the key transfer starts (followed by value transfer) and when everything is done, we confirm the topology calling `StateTransferManager.notifyEndOfStateTransfer`.